youtube_dl/extractor/neteasemusic.py

```diff
@@ -125,6 +133,8 @@ def extract_formats(self, info):
                         'filesize': int_or_none(song.get('size')),
                         'asr': int_or_none(details.get('sr')),
                     })
+        if not formats:
+            self.report_warning('Maybe you should use a proper proxy server located in China')
```

Comment:
If the site returns an error code (assume semantics like HTTP codes) and no formats, tell the user.

If no code and no formats, assume geo-blocking. The diagnostic will look like this:
```
ERROR: No media links found: probably this video is not available from your location due to geo restriction
This video is available in China.
You might want to use a VPN or a proxy server (with --proxy) to workaround.
```
If any other regions are served (HK? other E Asian countries?), their ISO 2-letter codes should be added to the list of countries.
```suggestion
                elif err == 0:
                    err = try_get(song, lambda x: x['code'], int)

        if not formats:
            msg = 'No media links found'
            if err != 0 and (err < 200 or err >= 400):
                raise ExtractorError(
                    '%s (site code %d)' % (msg, err, ), expected=True)
            else:
                self.raise_geo_restricted(
                    msg + ': probably this video is not available from your location due to geo restriction.',
                    countries=['CN'])

```